pkgs/development/python-modules/pybind11/default.nix

```diff
@@ -37,7 +37,11 @@ buildPythonPackage rec {
     "-DBoost_INCLUDE_DIR=${lib.getDev boost}/include"
     "-DEIGEN3_INCLUDE_DIR=${lib.getDev eigen}/include/eigen3"
     "-DBUILD_TESTING=on"
```

Comment:
When trying to cross compile `pybind11`, I noticed that `cmake`'s arguments include first `-DBUILD_TESTING=off` because it's a cross compilation, and then `-DBUILD_TESTING=on` due to this line. Since tests are disabled anyway in a cross compilation, do you think maybe you can merge https://github.com/NixOS/nixpkgs/pull/200845 into herek?

Reply:
BTW, you can add all of these cmakeFlags only if you cross compile, and you'll spare the massive rebuild.

Reply:
On a 2nd thought, let's keep this PR targeting `staging`, and let's update to [2.10.1](https://github.com/pybind/pybind11/releases/tag/v2.10.1) !

Reply:
For the update see https://github.com/NixOS/nixpkgs/pull/200338

Reply:
Making the flags conditional would just make it more likely for cross-compilation to break when the package is updated, so I'd like to avoid it even if it means going through staging.

I don't see why the update needs to be included in this PR, especially when there is an existing PR for it.

Reply:
> Making the flags conditional would just make it more likely for cross-compilation to break when the package is updated, so I'd like to avoid it even if it means going through staging.
>
> I don't see why the update needs to be included in this PR, especially when there is an existing PR for it.

You have a point..

Did you try my PR?